pandas/core/arrays/arrow/dtype.py

```diff
@@ -91,12 +140,9 @@ def _get_common_dtype(self, dtypes: list[DtypeObj]) -> DtypeObj | None:
             ]
```

Comment:
Something to note somewhere (rather for a follow-up) that the above will only work for pyarrow types that have a matching numpy dtype, which is not the case for all types (eg decimal, dictionary, date, nested types, etc)

Reply:
I defaulted the `numpy_dtype` to `np.dtype(object)` to those pyarrow types without a corresponding numpy type.

https://github.com/pandas-dev/pandas/pull/47034/files#diff-251a2430af30e9f685a07dbee6d5c024e832fc7e10569c7d43ee46bc115b422dR56

Reply:
I am not sure that will help (but without having tests that cover this it is hard to say). If `.numpy_dtype` returns object dtype, then the common dtype here will also be object dtype, and then `pa.from_numpy_dtype` below will fail (so this function will return None). That means that there can never be a proper `ArrowDtype` common dtype (that is not object dtype) for such arrow types. 

For example, if you have one array of decimal and one of float, the common dtype _could_ be float (not sure we want to do this, but let's assume for the example). With the current implementation, the `numpy_dtype` for those extension dtypes will be `np.dtype(object)` and `np.dtype(float)`. The common dtype for that will always be `np.dtype(object)`, which means that concatting such columns will result in a cast to object dtype, instead of casting to/preserving the float dtype.

So at some point, we should probably include pyarrow-specific logic in here that doesn't rely on converting to a numpy dtype and numpy's notion of a common type.

Reply:
> So at some point, we should probably include pyarrow-specific logic in here that doesn't rely on converting to a numpy dtype and numpy's notion of a common type.

Agreed, and makes sense that this shouldn't be `object` dtype in the long term. Would be great if this eventually follows pyarrow's type coercion rules if there is one :)